source/MaterialXGenShader/ColorManagementSystem.cpp

```diff
@@ -39,29 +39,31 @@ void ColorManagementSystem::loadLibrary(DocumentPtr document)
 
 bool ColorManagementSystem::supportsTransform(const ColorSpaceTransform& transform) const
 {
-    const string implName = getImplementationName(transform);
-    ImplementationPtr impl = _document->getImplementation(implName);
+    if (!_document)
+    {
+        throw ExceptionShaderGenError("No library loaded for color management system");
+    }
+    ImplementationPtr impl = getImplementation(transform);
     return impl != nullptr;
 }
 
 ShaderNodePtr ColorManagementSystem::createNode(const ShaderGraph* parent, const ColorSpaceTransform& transform, const string& name, 
                                                 GenContext& context) const
 {
-    const string implName = getImplementationName(transform);
-    ImplementationPtr impl = _document->getImplementation(implName);
+    ImplementationPtr impl = getImplementation(transform);
     if (!impl)
     {
         throw ExceptionShaderGenError("No implementation found for transform: ('" + transform.sourceSpace + "', '" + transform.targetSpace + "').");
     }
 
     // Check if it's created and cached already,
     // otherwise create and cache it.
-    ShaderNodeImplPtr nodeImpl = context.findNodeImplementation(implName);
+    ShaderNodeImplPtr nodeImpl = context.findNodeImplementation(impl->getName());
```

Comment:
Fix for handling targets which inherit from base ones like "arnold" which inherits from "genosl". Will check up inhertance list for parent targets to match. Fixes color management failing for these targets.